tesseract_motion_planners/ompl/src/ompl_motion_planner.cpp

```diff
@@ -104,439 +95,237 @@ bool OMPLMotionPlanner::terminate()
+  const std::size_t n_states = simple_setup->getSolutionPath().getStateCount();
+  if (n_states < request.seed.size())
   {
-    CONSOLE_BRIDGE_logError("In TrajOptPlannerUniversalConfig: env is a required parameter and has not been set");
-    return false;
+    // Upsample the number of states to match the length of the seed
+    simple_setup->getSolutionPath().interpolate(static_cast<unsigned>(request.seed.size()));
   }
+  else if (n_states > request.seed.size() && params.simplify)
+  {
+    // Simplify the solution
+    simple_setup->simplifySolution();
 
-  if (request.instructions.empty())
+    // Upsample the number of states to match the length of the seed if required
+    if (simple_setup->getSolutionPath().getStateCount() < request.seed.size())
+      simple_setup->getSolutionPath().interpolate(static_cast<unsigned>(request.seed.size()));
+  }
+  else
   {
-    CONSOLE_BRIDGE_logError("TrajOptPlannerUniversalConfig requires at least one instruction");
-    return false;
+    // Trajectory length matches or exceeds seed length and simplification is disabled, so no need to simplify or
+    // interpolate
   }
 
-  return true;
-}
+  // Get the results
+  response.data = std::static_pointer_cast<void>(simple_setup);
+  tesseract_common::TrajArray trajectory = toTrajArray(simple_setup->getSolutionPath(), extractor);
+  assert(checkStartState(simple_setup->getProblemDefinition(), trajectory.row(0), extractor));
+  assert(checkGoalState(simple_setup->getProblemDefinition(), trajectory.bottomRows(1).transpose(), extractor));
 
-OMPLProblem::Ptr createOMPLSubProblem(const PlannerRequest& request,
-                                      const tesseract_kinematics::JointGroup::ConstPtr& manip)
-{
-  auto sub_prob = std::make_unique<OMPLProblem>();
-  sub_prob->env = request.env;
-  sub_prob->env_state = request.env_state;
-  sub_prob->manip = manip;
-  sub_prob->contact_checker = request.env->getDiscreteContactManager();
-  sub_prob->contact_checker->setCollisionObjectsTransform(request.env_state.link_transforms);
-  sub_prob->contact_checker->setActiveCollisionObjects(manip->getActiveLinkNames());
-  return sub_prob;
-}
+  // Enforce limits
+  {
+    const std::string manipulator = request.instructions.getManipulatorInfo().manipulator;
+    auto joint_limits = request.env->getJointGroup(manipulator)->getLimits().joint_limits;
+    for (Eigen::Index i = 0; i < trajectory.rows(); i++)
+      tesseract_common::enforcePositionLimits(trajectory.row(i), joint_limits);
+  }
 
-std::vector<OMPLProblem::Ptr> OMPLMotionPlanner::createProblems(const PlannerRequest& request) const
-{
-  std::vector<OMPLProblem::Ptr> problem;
-  tesseract_kinematics::JointGroup::Ptr manip;
+  // Initialize the output by copying the input program and flattening it
+  response.results = request.seed;
+  std::vector<std::reference_wrapper<Instruction>> results_flattened =
+      flattenProgramToPattern(response.results, request.instructions);
```

Comment:
Is there a better/safer way to fill out the output trajectory? It's problematic to set the response equal to the seed and then overwrite the waypoints. I just fixed an issue where I was accidentally skipping the overwrite of the 0th element of the trajectory composite, but nothing failed and it wasn't obvious because there was already a default waypoint in that spot from the seed.